packages/block-editor/src/components/block-list/block.js

```diff
@@ -343,8 +344,62 @@ const applyWithDispatch = withDispatch( ( dispatch, ownProps, registry ) => {
 				getBlockAttributes,
 				getBlockName,
 				getBlockOrder,
+				getBlockIndex,
+				getBlockRootClientId,
+				canInsertBlockType,
 			} = registry.select( blockEditorStore );
 
+			function moveFirstItemUp( _clientId, changeSelection = true ) {
```

Comment:
A high-level comment stating the goal of the function and its general algorithm would be greatly appreciated. :)

Reply:
Done